module-apps/tests/widgets/test-DigitsContainer.cpp

```diff
@@ -11,7 +11,7 @@
 namespace
 {
 
-    template <size_t N> void checkDisplayedText(std::string text, const gui::DigitsContainer<N> &container)
+    void checkDisplayedText(std::string text, const gui::DigitsContainer &container)
```

Comment:
```suggestion
    void checkDisplayedText(const std::string& text, const gui::DigitsContainer &container)
```

Reply:
checkDisplayedText function modifies text so text cant be passed ass const & but I wll change to &&

Reply:
so then:
- function name is incorrect
- something is `no yes`

Reply:
What would be your proposal than for "ideal" name?

Reply:
I'd move the `pop` out of the function - if it has a checking purpose, it shouldn't alter the parameter - somehow related to single responsibility rule :P

Reply:
If you would remove pop out of the function please add comment related to pop move out of this function. I would disagree this function takse a Rvalue as parameter to check that has proper value. How it brakes single responsibility  ?

Reply:
```
The idea behind the SRP is that every class, module, or function in a program should have one responsibility/purpose in a program```
If the function is named `checkSomething` I'd say that it will do some testing and leave the parameters untouched. If it do anything on the arguments which are passed by reference/pointer, then it shouldn't be named this way, as it's confusing.

More over, I believe that this pop should be moved outside the function with the corresponding `std::for_each` - then you can have a function with testing-only, and rest of the test logic, which is responsible of shifting the text/popping it out/whatever

Reply:
You mentioned that SRP is broken but you did not mentioned these plural responsibilities. I assume that in your understanding this function does **check** text variable and  **modifies** text variable. However this function is not about checking text variable its about checking container on basis of text variable so in my opinion this solution is efficient and for sure does not break SRP. In another words in test environment checks what would be displayed on real device. I will not try to convenient you because we have different opinions/understanding in this topic and thats perfectly fine. 

Despite IMO this whole discussion is out of scope as in this commit  just changed function signature as DigitsContainer is not longer template (name, funcion parameters and body does not changed)and I have different opinion, I'm very grateful for your effort regarding aspiration for better quality code so plase provide the better implementaion containing your ideas. I will paste them in to this commit. It would speedup the review process and everyone will be satisfied.

Reply:
Apart from the `SRP` rule, it looks weird when we have the function `check` something and then modifies parameters inside. I agree with @SP2FET that you should at least change the function name. This small change can make this code  more readable.

Reply:
OK lets change the name than. @SP2FET and @dawidwojtas-mudita function name   doesDigitalsContainerWouldDislpayText would be more readable and would not confuse you? Please give your ideas if this name would not be good enough or lacking some way?

Reply:
TBH I can't find a proper name for this function, as it behaves quite strange for me - I'd say, that if we want to compare if the container's data is equal to the expected ones, we should just iterate through the container and verify it - the function just pops out the data if ithere's any, and then verifies if it's emptied out...

Reply:
or as a compromise - pass it via value and then modify the parameter inside the function only - I dug into the code more, and I see that it might have sense, so that's one of the solutions

Reply:
The naming is quite hard especially there is no precisely rules. Thanks for your try. Once again I encourage you to make proposals for better implementation allowing you easly name this function.

Reply:
I haven't seen your last comment while I posted my last comment. Currently in this review I did change signature to Rvalue reference it fits better because represents binding to temporary object(by default can be modified), but if you would like to pass text variable by copy its perfectly fine for me.

Reply:
let's leave it as it is - with the move instead of passing by ref - at least it's not confusing so much